src/native/corehost/hostpolicy/deps_format.cpp

```diff
@@ -417,19 +418,22 @@ void deps_json_t::process_runtime_targets(const json_parser_t::value_t& json, co
                     version_t::parse(file_version_str, &file_version);
                 }
 
+                const pal::string_t& local_path = get_optional_path(file.value, _X("localPath"));
```

Comment:
Looking at `get_optional_path()`, I think this is incorrect. It returns a `std::string_t`, which means once we step over this that will be freed and the reference will be dangling. It is entirely possible this is benefiting from the `const &` rules, but it really should just be `pal::string_t` rather than a reference. For all of them.

Reply:
I did confirm the temporary is being kept alive by the `const&`, but I think it would be less error prone to just use the return value type.